openmc/mgxs_library.py

```diff
@@ -970,144 +1095,202 @@ def set_multiplicity_mgxs(self, nuscatter, scatter=None, nuclide='total',
+                    g_out_bounds[g_in, 0] = nz[0][0]
+                    g_out_bounds[g_in, 1] = nz[0][-1]
+                # Now create the flattened scatter matrix array
+                matrix = self._scatter_matrix[i]
+                flat_scatt = []
+                for g_in in range(G):
+                    for g_out in range(g_out_bounds[g_in, 0],
+                                       g_out_bounds[g_in, 1] + 1):
+                        for l in range(len(matrix[:, g_in, g_out])):
+                            flat_scatt.append(matrix[l, g_in, g_out])
+                # And write it.
+                scatt_grp = xs_grp.create_group('scatter data')
+                scatt_grp.create_dataset("scatter matrix",
+                                         data=np.array(flat_scatt))
+                # Repeat for multiplicity
+                if self._multiplicity_matrix[i] is not None:
+                    # Now create the flattened scatter matrix array
+                    matrix = self._multiplicity_matrix[i][:, :]
+                    flat_mult = []
+                    for g_in in range(G):
+                        for g_out in range(g_out_bounds[g_in, 0],
+                                           g_out_bounds[g_in, 1] + 1):
+                            flat_mult.append(matrix[g_in, g_out])
+                    scatt_grp.create_dataset("multiplicity matrix",
+                                             data=np.array(flat_mult))
+
+                # And finally, adjust g_out_bounds for 1-based group counting
+                # and write it.
+                g_out_bounds[:, :] += 1
+                scatt_grp.create_dataset("g_min", data=g_out_bounds[:, 0])
+                scatt_grp.create_dataset("g_max", data=g_out_bounds[:, 1])
+
+            elif self.representation == 'angle':
+                Np = self.num_polar
+                Na = self.num_azimuthal
+                g_out_bounds = np.zeros((Np, Na, G, 2), dtype=np.int)
+                for p in range(Np):
+                    for a in range(Na):
+                        for g_in in range(G):
+                            matrix = self._scatter_matrix[i][p, a, 0, g_in, :]
+                            nz = np.nonzero(matrix)
+                            g_out_bounds[p, a, g_in, 0] = nz[0][0]
+                            g_out_bounds[p, a, g_in, 1] = nz[0][-1]
+                # Now create the flattened scatter matrix array
+                flat_scatt = []
+                for p in range(Np):
+                    for a in range(Na):
+                        matrix = self._scatter_matrix[i][p, a, :, :, :]
+                        for g_in in range(G):
+                            for g_out in range(g_out_bounds[p, a, g_in, 0],
+                                               g_out_bounds[p, a, g_in, 1] + 1):
+                                for l in range(len(matrix[:, g_in, g_out])):
+                                    flat_scatt.append(matrix[l, g_in, g_out])
```

Comment:
That won't allow me to get rid of the zeros between our g_min and g_max from the data, so I chose against it.